### ISSUE TITLE
Test did:key with Ed25519 and X25519 verification method keys in single DID doc

### ISSUE THREAD

**Issue by frankhinek** [TBD54566975/ssi-sdk#334]
Currently the tests include the [W3C `did:key` test vectors](https://w3c-ccg.github.io/did-method-key/#test-vectors):

https://github.com/TBD54566975/ssi-sdk/blob/0ce55b2b026291bb2cec435aa88fffadd89a2332/did/key_test.go#L352-L377

But don't fully test the Ed25519/X25519 scenario in which a single DID document has both types of keys present:
https://github.com/w3c-ccg/did-method-key/blob/main/test-vectors/ed25519-x25519.json

In other words, `verificationMethod` with length 2, one of which is Ed25519 and the other X25519.

No particular reason the test shouldn't pass.

**Comment by decentralgabe** [comment]
we have https://github.com/TBD54566975/ssi-sdk/issues/176 for interop with the did key test suite - which we should keep open.

For this, it seems like we should update did:key to support showing two keys when adding x25519.

**Comment by andresuribe87** [comment]
FWIW this is enabled in ssi-service. Perhaps we should simply move this into the sdk. 

https://github.com/TBD54566975/ssi-service/blob/96b09466c6a1a7080f802f3e986f7652b59cc1ed/internal/did/access.go#L25-L51

**Comment by decentralgabe** [comment]
good callout - agree we should move it into the sdk
